tensorflow/compiler/xla/service/gpu/tests/gpu_index_test.cc

```diff
@@ -105,13 +105,24 @@ TEST_F(GpuIndexTest, CompatibleUseLinearIndexWithReshapeAndBroadcast) {
                     .ValueOrDie();
 
   // Check the optimized IR reuses the linear index by calculating modulo 14.
-  CompileAndVerifyIr(std::move(module),
-                     R"(
+
+  // In the IR generated for AMDGPUs, we do not seem to have the
+  // the addrspace(1) attribute for the lines being checked by the following
+  // patterns still need to investigate why that is the case, and whether or not
+  // it is ok
+  auto expected_ir = is_built_with_rocm_ ? R"(
+; CHECK: %[[urem1:.*]] = urem i{{[0-9]*}} %[[linear_index:.*]], 14
+; CHECK: %[[bitcast:.*]] = bitcast i8* %[[alloc:.*]] to float*
```

Comment:
It's probably not worth it to duplicate the pattern to catch addrspace cast. Could you use a regexp to optionally match and addrspace cast instead?

Reply:
will push this change out in the next commit as well.